clash_lib/src/app/dispatcher/dispatcher_impl.rs

```diff
@@ -22,17 +22,18 @@ use std::{
 };
 use tokio::{io::AsyncWriteExt, sync::RwLock, task::JoinHandle};
 use tracing::{Instrument, debug, error, info, info_span, instrument, trace, warn};
-
-use crate::app::dns::ThreadSafeDNSResolver;
+use watfaq_resolver::{AbstractResolver, Resolver};
+use watfaq_state::Context;
 
 use super::statistics_manager::Manager;
 
 const DEFAULT_BUFFER_SIZE: usize = 16 * 1024;
 
 pub struct Dispatcher {
+    ctx: Arc<Context>,
```

Comment:
same here

Reply:
The `Context` is what we talked before. It is another solution of global values
https://github.com/Watfaq/clash-rs/pull/730#issuecomment-2676542814

Reply:
Yeah I know that.

We shouldn't add too many changes in one PR